### PR TITLE
[WIP] pythonPackages.cq-editor: init at 0.1RC1

### PR DESCRIPTION
**Pull request by costrouc**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by costrouc** [comment]
Nearly complete and working with the cq-editor but broken for now.

File: pkgs/development/python-modules/cadquery/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pyparsing
+, opencascade_oce
+, pythonocc-core
```
Comment (marcus7070) [inline pkgs/development/python-modules/cadquery/default.nix]: @costrouc, you've probably already noticed this, but CadQuery use [their own branch of pythonocc-core](https://github.com/CadQuery/pythonocc-core) and the current one from nixpkgs won't work.

**Comment by marcus7070** [comment]
@costrouc, I've made some progress in [my own fork](https://github.com/marcus7070/nixpkgs/tree/python-cq-editor-init). I thought I could submit a pull request to costrouc/nixpkgs, but Github won't let me select yours (perhaps there are too many forks of nixpkgs). 

Currently struggling with qtconsole not finding any pyqt version. 

```
[nix-shell:~/tempbuild/nixpkgs]$ cq-editor
Traceback (most recent call last):
  File "/nix/store/in688ijxjgw6ndqkkq92m1y8r6szkdj5-cq-editor-0.1RC1/bin/.cq-editor-wrapped", line 7, in <module>
    from cq_editor.__main__ import main
  File "/nix/store/in688ijxjgw6ndqkkq92m1y8r6szkdj5-cq-editor-0.1RC1/lib/python3.7/site-packages/cq_editor/__main__.py", line 12, in <module>
    from .main_window import MainWindow
  File "/nix/store/in688ijxjgw6ndqkkq92m1y8r6szkdj5-cq-editor-0.1RC1/lib/python3.7/site-packages/cq_editor/main_window.py", line 14, in <module>
    from .widgets.console import ConsoleWidget
  File "/nix/store/in688ijxjgw6ndqkkq92m1y8r6szkdj5-cq-editor-0.1RC1/lib/python3.7/site-packages/cq_editor/widgets/console.py", line 3, in <module>
    from qtconsole.rich_jupyter_widget import RichJupyterWidget
  File "/nix/store/5pygklssig4piyfrdc9sjmdwjclvgqn2-python3.7-qtconsole-4.5.1/lib/python3.7/site-packages/qtconsole/rich_jupyter_widget.py", line 9, in <module>
    from qtconsole.qt import QtCore, QtGui
  File "/nix/store/5pygklssig4piyfrdc9sjmdwjclvgqn2-python3.7-qtconsole-4.5.1/lib/python3.7/site-packages/qtconsole/qt.py", line 23, in <module>
    QtCore, QtGui, QtSvg, QT_API = load_qt(api_opts)
  File "/nix/store/5pygklssig4piyfrdc9sjmdwjclvgqn2-python3.7-qtconsole-4.5.1/lib/python3.7/site-packages/qtconsole/qt_loaders.py", line 387, in load_qt
    api_options))
ImportError: 
    Could not load requested Qt binding. Please ensure that
    PyQt4 >= 4.7, PyQt5, PySide >= 1.0.3 or PySide2 is available,
    and only one is imported per session.

    Currently-imported Qt library:   'pyqt5'
    PyQt4 installed:                 False
    PyQt5 installed:                 False
    PySide >= 1.0.3 installed:       False
    PySide2 installed:               False
    Tried to load:                   ['pyqt5']
```

**Comment by costrouc** [comment]
@marcus7070 since this PR is most important to you do you mind opening a PR for this package? I will close this one. I am happy to give you any help you might need. I will say that the hardest part of this packaging is qt (which I don't have much experience with). 

Let me know if this is okay with you and I'll close this PR in favor of yours

**Comment by marcus7070** [comment]
@costrouc Sure. I'm going to start an issue for this instead of going straight to a pull request, just in case I hand it off to someone else again.

**Comment by costrouc** [comment]
Great saw that you opened a PR. Thanks! I will close this one

**Pull request closed without merging**